From 553c74a7babafa2803f739dcb01162715a6c36ca Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Tue, 20 Jan 2015 09:32:43 -0500
Subject: [PATCH 084/315] session: start user X sessions on a new VT

We now have all the plumbing in place to start user X sessions
on new VTs, so we should do it.

This commit does that.

https://bugzilla.gnome.org/show_bug.cgi?id=744764
---
 daemon/gdm-session.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/daemon/gdm-session.c b/daemon/gdm-session.c
index 17264903..0edfaa14 100644
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -2895,6 +2895,13 @@ gdm_session_get_display_mode (GdmSession *self)
         }
 #endif
 
+        /* Non-seat0 sessions share their X server with their login screen
+         * for now.
+         */
+        if (g_strcmp0 (self->priv->display_seat_id, "seat0") != 0) {
+                return GDM_SESSION_DISPLAY_MODE_REUSE_VT;
+        }
+
         /* The X session used for the login screen uses the
          * X server started up by the slave, so it should be
          * reuse VT
@@ -2903,9 +2910,10 @@ gdm_session_get_display_mode (GdmSession *self)
                 return GDM_SESSION_DISPLAY_MODE_REUSE_VT;
         }
 
-        /* X sessions are for now ran in classic mode where
-         * we reuse the existing greeter. */
-        return GDM_SESSION_DISPLAY_MODE_REUSE_VT;
+        /* user based X sessions start on a new VT now and are managed
+         * by logind
+         */
+        return GDM_SESSION_DISPLAY_MODE_LOGIND_MANAGED;
 }
 
 void
-- 
2.15.1

